x/observer/types/messages_add_block_header.go

```diff
@@ -51,33 +49,23 @@ func (msg *MsgAddBlockHeader) GetSignBytes() []byte {
 func (msg *MsgAddBlockHeader) ValidateBasic() error {
 	_, err := sdk.AccAddressFromBech32(msg.Creator)
 	if err != nil {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+		return cosmoserrors.Wrapf(sdkerrors.ErrInvalidAddress, err.Error())
 	}
 
-	if len(msg.BlockHash) > 32 {
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid msg.txhash; too long (%d)", len(msg.BlockHash))
-	}
-	if len(msg.BlockHeader) > 1024 { // on ethereum the block header is ~538 bytes in RLP encoding
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid msg.blockheader; too long (%d)", len(msg.BlockHeader))
-	}
 	if common.IsEthereum(msg.ChainId) {
-		// RLP encoded block header
-		var header types.Header
-		err = rlp.DecodeBytes(msg.BlockHeader, &header)
-		if err != nil {
-			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid block header; cannot decode RLP (%s)", err)
-		}
-		if err = header.SanityCheck(); err != nil {
-			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid block header; sanity check failed (%s)", err)
-		}
-		if bytes.Compare(msg.BlockHash, header.Hash().Bytes()) != 0 {
-			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid block header; tx hash mismatch")
-		}
-		if msg.Height != header.Number.Int64() {
-			return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "invalid block header; height mismatch")
+		if len(msg.BlockHash) > 32 {
```

Comment:
I would keep previous `if len(msg.BlockHeader) > 1024` check to prevent malicious zetaclient feeding illegal (huge) blocker headers and cause problems.